### PR TITLE
Chevron position for linked image without caption

### PR DESCRIPTION
**Pull request by vforge**
https://wikia-inc.atlassian.net/browse/HG-533

### CODE REVIEW

File: front/templates/main/components/linked-gallery-media.hbs [hunk 1]
```diff
@@ -1,14 +1,20 @@
 {{#each media in limitedMedia}}
 	<figure class='linked-gallery-{{unbound media.type}}'>
```
Comment (rogatty) [inline front/templates/main/components/linked-gallery-media.hbs]: How about `<figure class='linked-gallery-{{unbound media.type}}{{#if media.caption}} captioned{{/if}}'>`?

File: front/templates/main/components/linked-gallery-media.hbs [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+				{{#if media.caption}}
+					{{svg 'chevron-linked' viewBox='0 0 20 20' class='chevron moved'}}
+				{{else}}
+					{{svg 'chevron-linked' viewBox='0 0 20 20' class='chevron'}}
+				{{/if}}
 			</a>
 		{{else}}
 			<img {{bind-attr src=media.thumbUrl}} {{bind-attr class='media.load:loaded'}} data-gallery-ref='{{unbound media.galleryRef}}'>
 		{{/if}}
-		<figcaption>{{{media.caption}}}</figcaption>
```
Comment (rogatty) [inline front/templates/main/components/linked-gallery-media.hbs]: Please ask @dianafa why she didn't do it like that while working on https://wikia-inc.atlassian.net/browse/HG-495. I think there was some reason.

File: front/templates/main/components/linked-gallery-media.hbs (hunk 1 above)
Comment (vforge) [inline front/templates/main/components/linked-gallery-media.hbs]: I want to create common styling/behaviour and there's no ```<figure>``` in ```image-media.hbs```

Comment (rogatty) [inline front/templates/main/components/linked-gallery-media.hbs]: `<figure>` element in `ImageMediaComponent` is inherited from `MediaComponent` where it is the main element (see it here https://github.com/Wikia/mercury/blob/HG-533/front/scripts/main/components/MediaComponent.ts#L9). This means that you can use http://emberjs.com/api/classes/Ember.Component.html#property_classNameBindings in the `ImageMediaComponent.ts` file to add `captioned` class to `<figure>` element.

File: front/styles/module/article/_media.scss [hunk 3]
```diff
@@ -27,11 +27,15 @@
...
 		bottom: 0;
 		height: 20px;
-		margin-bottom: 50px;
+		margin-bottom: 0;
 		position: absolute;
 		right: 0;
 		width: 20px;
 		z-index: $z-1;
+
+		&.moved {
```
Comment (kenkouot) [inline front/styles/module/article/_media.scss]: can you give a little context on this subclass?

Comment (vforge) [inline front/styles/module/article/_media.scss]: I'll sove this differently, with Igor's suggestion, I'll add class to ```<figure>```

File: front/templates/main/components/linked-gallery-media.hbs (hunk 2 above)
Comment (dianafa) [inline front/templates/main/components/linked-gallery-media.hbs]: explained :)

File: front/styles/module/article/_media.scss [hunk 4]
```diff
@@ -27,12 +27,16 @@
 		background-color: #fff;
 		bottom: 0;
 		height: 20px;
-		margin-bottom: 50px;
+		margin-bottom: 0;
```
Comment (rogatty) [inline front/styles/module/article/_media.scss]: Is it necessary? Wouldn't it be 0 by default?

File: front/scripts/main/components/GalleryMediaComponent.ts [hunk 5]
```diff
@@ -23,6 +24,10 @@ App.GalleryMediaComponent = App.MediaComponent.extend({
 		this.get('media').forEach((media: ArticleMedia, index: number) => {
 			media.galleryRef = index;
 			media.thumbUrl = emptyGif;
+			media.classNames = 'linked-gallery-' + media.type;
```
Comment (rogatty) [inline front/scripts/main/components/GalleryMediaComponent.ts]: Not every gallery is a linked gallery. That's why there is `LinkedGalleryMediaComponent`.

File: front/scripts/main/components/GalleryMediaComponent.ts [hunk 6]
```diff
@@ -23,6 +24,10 @@ App.GalleryMediaComponent = App.MediaComponent.extend({
...
+			if (media.caption) {
+				media.classNames += ' has-caption';
```
Comment (rogatty) [inline front/scripts/main/components/GalleryMediaComponent.ts]: How about pushing to array? If Handlebars won't render it properly by default maybe we should have a helper that will convert it to string.

File: front/scripts/main/components/ImageMediaComponent.ts [hunk 7]
```diff
@@ -67,6 +67,12 @@ App.ImageMediaComponent = App.MediaComponent.extend({
 			'height:%@px;'.fmt(this.get('computedHeight'));
 	}.property('computedHeight', 'visible'),
 
+	hasCaption: function (): boolean {
+		var caption = this.get('media').caption;
+
+		return caption && typeof caption === 'string' && caption.length > 0
```
Comment (rogatty) [inline front/scripts/main/components/ImageMediaComponent.ts]: I believe a correct type should be guaranteed by the API, so `caption.length > 0` should be enough.
@hakubo what do you think? Should we do validation like that? If yes then there is a lot to add in the existing code.

Also semicolon is missing.

Comment (rogatty) [inline front/scripts/main/components/ImageMediaComponent.ts]: Or just use http://emberjs.com/api/classes/Ember.html#method_computed_notEmpty

File: front/templates/main/components/linked-gallery-media.hbs [hunk 8]
```diff
@@ -1,14 +1,16 @@
 {{#each media in limitedMedia}}
-	<figure class='linked-gallery-{{unbound media.type}}'>
+	<figure class='linked-gallery-{{unbound media.type}} {{unbound media.captionClass}}'>
```
Comment (rogatty) [inline front/templates/main/components/linked-gallery-media.hbs]: The last thing - I would remove space from here and put it in the `captionClass` property. This way we won't have it when not necessary.

**Comment by rogatty** [comment]
Merge the latest dev to your branch, please.

File: front/scripts/main/components/ImageMediaComponent.ts [hunk 9]
```diff
@@ -4,12 +4,13 @@
 
 App.ImageMediaComponent = App.MediaComponent.extend({
 	classNames: ['article-image'],
-	classNameBindings: ['visible'],
+	classNameBindings: ['hasCaption:has-caption', 'visible'],
```
Comment (hakubo) [inline front/scripts/main/components/ImageMediaComponent.ts]: hasCaption should be dasherized by default

**Comment by vforge** [comment]
@rogatty Done and done.

**Comment by rogatty** [comment]
LGTM

**Pull request merged**